driver/mounter.go

```diff
@@ -157,21 +157,34 @@ func (m *mounter) Format(source, fsType string) error {
 		mkfsArgs = []string{"-F", source}
 	}
 
-	m.log.WithFields(logrus.Fields{
-		"cmd":  mkfsCmd,
-		"args": mkfsArgs,
-	}).Info("executing format command")
+	if !luksContext.EncryptionEnabled {
+		m.log.WithFields(logrus.Fields{
+			"cmd":  mkfsCmd,
+			"args": mkfsArgs,
+		}).Info("executing format command")
 
-	out, err := exec.Command(mkfsCmd, mkfsArgs...).CombinedOutput()
-	if err != nil {
-		return fmt.Errorf("formatting disk failed: %v cmd: '%s %s' output: %q",
-			err, mkfsCmd, strings.Join(mkfsArgs, " "), string(out))
-	}
+		out, err := exec.Command(mkfsCmd, mkfsArgs...).CombinedOutput()
+		if err != nil {
+			return fmt.Errorf("formatting disk failed: %v cmd: '%s %s' output: %q",
+				err, mkfsCmd, strings.Join(mkfsArgs, " "), string(out))
+		}
+		return nil
+	} else {
+		err := luksContext.validate()
```

Comment:
Are you able to return a formatted error instead using fmt.Errorf, stating something like `validation failed : %s`?

Reply:
sure, updated